check_run/overrides/payment_entry.py

```diff
@@ -53,6 +57,144 @@ def get_valid_reference_doctypes(self):
 		elif self.party_type == "Employee":
 			return ("Journal Entry", "Expense Claim")  # Expense Claim
 
+	# validation msg improvement
+	def validate_allocated_amount(self):
+		if self.payment_type == "Internal Transfer":
+			return
+
+		if self.party_type in ("Customer", "Supplier"):
+			self.validate_allocated_amount_with_latest_data()
+		else:
+			fail_message = _(
+				"{0} Row {1} / {2}: Allocated Amount of {3} cannot be greater than outstanding amount of {4}."
+			)
+			for d in self.get("references"):
+				if (flt(d.allocated_amount)) > 0 and flt(d.allocated_amount) > flt(d.outstanding_amount):
+					frappe.throw(
+						fail_message.format(
+							self.party_name,
+							d.idx,
+							get_link_to_form(d.reference_doctype, d.reference_name),
+							d.allocated_amount,
+							d.outstanding_amount,
+						)
+					)
+
+				# Check for negative outstanding invoices as well
+				if flt(d.allocated_amount) < 0 and flt(d.allocated_amount) < flt(d.outstanding_amount):
+					frappe.throw(
+						fail_message.format(
+							self.party_name,
+							d.idx,
+							get_link_to_form(d.reference_doctype, d.reference_name),
+							d.allocated_amount,
+							d.outstanding_amount,
+						)
+					)
+
+	def validate_allocated_amount_with_latest_data(self):
+		if self.references:
+			uniq_vouchers = {(x.reference_doctype, x.reference_name) for x in self.references}

```

Comment:
This is a matter of personal taste, but I'm not a big fan of this kind of abbreviated variable naming. In this context, I'd argue that `vouchers` is fine, `all_vouchers` would be good if there's a concern about reusing `vouchers` (mypy might complain about reassignment and that's how you ended up here). `unique_vouchers` is only two characters longer and improves readability by being a complete and recognizable word, where "uniq" is not.

Reply:
I have rename variable 'uniq_vouchers' to 'unique_vouchers' in override payment_entry.py